src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -0,0 +1,196 @@
+using Bit.App.Resources;
+using Bit.App.Utilities;
+using Bit.Core;
+using Bit.Core.Abstractions;
+using Bit.Core.Enums;
+using Bit.Core.Models.Domain;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public class LoginPasswordlessRequestViewModel : CaptchaProtectedViewModel
+    {
+        private const int REQUEST_TIME_UPDATE_PERIOD_IN_SECONDS = 4;
+
+        private IDeviceActionService _deviceActionService;
+        private IAuthService _authService;
+        private ISyncService _syncService;
+        private II18nService _i18nService;
+        private IStateService _stateService;
+        private ICryptoFunctionService _cryptoFunctionService;
+        private ICryptoService _cryptoService;
+        private IPasswordGenerationService _passwordGenerationService;
+        private IPlatformUtilsService _platformUtilsService;
+        private IEnvironmentService _environmentService;
+        private ILogger _logger;
+
+        protected override II18nService i18nService => _i18nService;
+        protected override IEnvironmentService environmentService => _environmentService;
+        protected override IDeviceActionService deviceActionService => _deviceActionService;
+        protected override IPlatformUtilsService platformUtilsService => _platformUtilsService;
+
+        private CancellationTokenSource _checkLoginRequestStatusCts;
+        private Task _checkLoginRequestStatusTask;
+        private string _fingerprintPhrase;
+        private string _email;
+        private string _requestId;
+        private string _requestAccessCode;
+        private Tuple<byte[], byte[]> _requestKeyPair;
+
+        public LoginPasswordlessRequestViewModel()
+        {
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>();
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>();
+            _environmentService = ServiceContainer.Resolve<IEnvironmentService>();
+            _authService = ServiceContainer.Resolve<IAuthService>();
+            _syncService = ServiceContainer.Resolve<ISyncService>();
+            _i18nService = ServiceContainer.Resolve<II18nService>();
+            _stateService = ServiceContainer.Resolve<IStateService>();
+            _cryptoFunctionService = ServiceContainer.Resolve<ICryptoFunctionService>();
+            _cryptoService = ServiceContainer.Resolve<ICryptoService>();
+            _passwordGenerationService = ServiceContainer.Resolve<IPasswordGenerationService>();
+            _logger = ServiceContainer.Resolve<ILogger>();
+
+            PageTitle = AppResources.LogInWithAnotherDevice;
+
+            CreatePasswordlessLoginCommand = new AsyncCommand(() => Device.InvokeOnMainThreadAsync(CreatePasswordlessLoginAsync),
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
```

Comment:
If possible, move the invoking to the main thread on the parts actually needed inside `CreatePasswordlessLoginAsync`. E.g. the call to `await _authService.PasswordlessCreateLoginRequestAsync(...)` shouldn't happen in the main thread

Reply:
@andrebispo5 did you mark this as resolved because it's not possible to move the invocation to the main thread to inside the method's parts where it's needed?

Reply:
I'm sorry, i was doing some trial and error changes and ended up confusing myself. It's resolved now, promise!